### PR TITLE
feat: add `math/base/special/nanmax`

### PR DESCRIPTION
**Pull request by RidamGarg**
Generated The PR for issue #2317

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/nanmax/README.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+
+If both argument are `NaN`, the function returns `NaN`.
+
+```javascript
+var v = nanmax( NaN, NaN );
+// returns NaN
+
+```
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/README.md]:
````suggestion
If both arguments are `NaN`, the function returns `NaN`.

```javascript
var v = nanmax( NaN, NaN );
// returns NaN
```

````

File: lib/node_modules/@stdlib/math/base/special/nanmax/README.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+</section>
+
+<!-- /.related -->
+
+<!-- Section for all links. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->
+
+<section class="links">
+
+
+
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/README.md]:
```suggestion
<!-- <related-links> -->

<!-- </related-links> -->
```

File: lib/node_modules/@stdlib/math/base/special/nanmax/examples/index.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+// returns 4.0
+console.log(m);
+
+m = nanmax( 4.0, NaN );
+// returns 4.0
+console.log(m);
+
+m = nanmax( NaN, NaN );
+// returns NaN
+console.log(m);
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/examples/index.js]:
```suggestion
var m = nanmax( 3.0, 4.0 );
console.log( m );
// => 4.0

m = nanmax( NaN, 4.0 );
console.log( m );
// => 4.0

m = nanmax( 4.0, NaN );
console.log( m );
// => 4.0

m = nanmax( NaN, NaN );
console.log( m );
// => NaN
```

File: lib/node_modules/@stdlib/math/base/special/nanmax/package.json [hunk 4]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@stdlib/math/base/special/nanmax",
+  "version": "0.0.0",
+  "description": "Returns the maximum value, ignoring NaN.",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/package.json]:
```suggestion
  "description": "Return the maximum value, ignoring NaN.",
```

File: lib/node_modules/@stdlib/math/base/special/nanmax/package.json [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    "windows"
+  ],
+  "keywords": [
+    "stdlib",
+    "stdmath",
+    "mathematics",
+    "math",
+    "math.min",
+    "minimum",
+    "min",
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/package.json]:
```suggestion
    "math.max",
    "maximum",
    "max",
```

File: lib/node_modules/@stdlib/math/base/special/nanmax/lib/main.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+// MODULES //
+
+var max = require( '@stdlib/math/base/special/max');
+var isnan = require( '@stdlib/math/base/assert/is-nan');
+
+
+// MAIN //
+
+/**
+* Return the maximum value, ignoring NaN.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/lib/main.js]:
```suggestion
* Returns the maximum value, ignoring NaN.
```

File: lib/node_modules/@stdlib/math/base/special/nanmax/lib/main.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+*
+* @param {number} x - first number
+* @param {number} y - second number
+* @returns {number} minimum value
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/math/base/special/nanmax/lib/main.js]:
```suggestion
* @returns {number} maximum value
```

**Review by Planeshifter — APPROVED** [review]
Thank you @RidamGarg for this PR! 👍 

PR looked good; just a few copy-paste and formatting errors which I fixed. Will merge once CI checks have passed.

For future reference: fixes like the ones you included for `nanmin` should be made in separate PRs / branches to ensure a clean git commit history and ensure correctness of the automatically generated changelogs (see e.g. the [one for nanmin](https://github.com/stdlib-js/math-base-special-nanmin/blob/main/CHANGELOG.md). I manually made the necessary changes in `nanmin`, so it's all taken care of, but please keep in mind in the future.

File: lib/node_modules/@stdlib/math/base/special/nanmax/docs/types/index.d.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+// TypeScript Version: 4.1
+
+/**
+* Return the maximum value, ignoring NaN.
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmax/docs/types/index.d.ts:22]: Returns

File: lib/node_modules/@stdlib/math/base/special/nanmax/benchmark/benchmark.js [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var values;
+	var x;
+	var y;
+	var z;
+	var i;
+
+	values = [ 3, 2, 2.0, 1.5, NaN ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/nanmax/benchmark/benchmark.js:38]: 3.0, 2.0

Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/nanmax/benchmark/benchmark.js:38]: Changed.

File: lib/node_modules/@stdlib/math/base/special/nanmax/docs/types/index.d.ts (hunk 8 above)
Comment (Planeshifter) [reply lib/node_modules/@stdlib/math/base/special/nanmax/docs/types/index.d.ts:22]: Changed.

**Pull request merged**